stores/sql/main.go

```diff
@@ -187,33 +186,43 @@ func ArchiveContract(ctx context.Context, tx sql.Tx, fcid types.FileContractID,
 	return nil
 }
 
-func Autopilot(ctx context.Context, tx sql.Tx, id string) (api.Autopilot, error) {
-	row := tx.QueryRow(ctx, "SELECT identifier, config, current_period FROM autopilots WHERE identifier = ?", id)
-	ap, err := scanAutopilot(row)
+func AutopilotState(ctx context.Context, tx sql.Tx) (cfg api.AutopilotState, err error) {
+	err = tx.QueryRow(ctx, `
+SELECT
+	current_period,
+	contracts_set,
+	contracts_amount,
+	contracts_period,
+	contracts_renew_window,
+	contracts_download,
+	contracts_upload,
+	contracts_storage,
+	contracts_prune,
+	hosts_allow_redundant_ips,
+	hosts_max_downtime_hours,
+	hosts_min_protocol_version,
+	hosts_max_consecutive_scan_failures
+FROM autopilot_state
+WHERE id = ?`, sql.AutopilotStateID).Scan(
+		&cfg.CurrentPeriod,
+		&cfg.Contracts.Set,
+		&cfg.Contracts.Amount,
+		&cfg.Contracts.Period,
+		&cfg.Contracts.RenewWindow,
+		&cfg.Contracts.Download,
+		&cfg.Contracts.Upload,
+		&cfg.Contracts.Storage,
+		&cfg.Contracts.Prune,
+		&cfg.Hosts.AllowRedundantIPs,
+		&cfg.Hosts.MaxDowntimeHours,
+		&cfg.Hosts.MinProtocolVersion,
+		&cfg.Hosts.MaxConsecutiveScanFailures,
+	)
 	if errors.Is(err, dsql.ErrNoRows) {
-		return api.Autopilot{}, api.ErrAutopilotNotFound
-	} else if err != nil {
-		return api.Autopilot{}, fmt.Errorf("failed to fetch autopilot: %w", err)
-	}
-	return ap, nil
-}
-
-func Autopilots(ctx context.Context, tx sql.Tx) ([]api.Autopilot, error) {
-	rows, err := tx.Query(ctx, "SELECT identifier, config, current_period FROM autopilots")
-	if err != nil {
-		return nil, fmt.Errorf("failed to fetch autopilots: %w", err)
+		_, err = tx.Exec(ctx, "INSERT INTO autopilot_state (id, created_at, current_period) VALUES (?, ?, 0);", sql.AutopilotStateID, time.Now())
```

Comment:
This is a bit fragile since it is only happening in `AutopilotState`. If you call another method first (e.g. to update the state or period) you might be screwed. 

Imo a better way would be to have a dedicated `InitAutopilotState` method which the autopilot calls on startup and then we are essentially good everywhere.

Reply:
I changed it. I copied the behaviour of `consensus_infos` to stay somewhat consistent but for settings we do something else entirely. One thing I don't like with an `Init` is that we rely on it being there after it was initialised. Which is fine, just wanted to draw attention to the fact I'm not checking for `ErrNoRows` and return autopilot not found anywhere.